cmd/kubeadm/app/master/manifests.go

```diff
@@ -234,11 +234,19 @@ func k8sVolume(cfg *kubeadmapi.MasterConfiguration) api.Volume {
 	}
 }
 
-func k8sVolumeMount() api.VolumeMount {
+func k8sVolumeMount(readOnly bool) api.VolumeMount {
```

Comment:
If `/etc/kubernetes` is read only, then the PKI volume cannot be mounted. This makes it optional to mount as read only, and is only false for the self-hosted controller-manager.